src/imap/client.rs

```diff
@@ -140,19 +136,17 @@ impl Client {
     }
 
     pub async fn connect_secure_socks5(
-        target_addr: &ServerAddress,
+        target_addr: impl net::ToSocketAddrs,
+        domain: &str,
         strict_tls: bool,
         socks5_config: Socks5Config,
     ) -> Result<Self> {
-        let socks5_stream: Box<dyn SessionStream> = Box::new(
```

Comment:
Btw, can we make rustfmt less intrusive? The purpose of line wrapping is not only to make a code more readable, but usually also to reduce diffs. Currently i don't see how that's addressed